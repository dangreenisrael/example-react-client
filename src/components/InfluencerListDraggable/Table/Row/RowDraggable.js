import React, {Component} from 'react';
import {DragSource} from 'react-dnd';
import RowCheckboxContainer from './RowCheckboxContainer';
import Tag from '../../../Tag'

const boxSource = {
  beginDrag(props) {
    return { id: props.id };
  },
};

@DragSource(props => "application", boxSource, (connect, monitor) => ({
  connectDragSource: connect.dragSource(),
  connectDragPreview: connect.dragPreview(),
  isDragging: monitor.isDragging(),
}))
export default class DraggableRow extends Component {
  render() {
    this.props.connectDragPreview(this.props.dragPreview);
    const {activeStyle, connectDragSource, activeColumn, data, headings, id, selected, removeTag} = this.props;
    const style = selected ? activeStyle : {};
    return connectDragSource(
        <tr style={style}>
          <td>
            <RowCheckboxContainer rowId={id}/>
          </td>
          {headings.map((heading, i)=>{
            const propName = heading.propertyName;
            const style = (propName === activeColumn) ? activeStyle: {};
            return <td style={style} key={`data-${i}`}>
              {data[propName]}
            </td>
          })}
          <td>
            {data.tags.map((tag, i)=>{
              return <Tag key={`tags-${i}`} title={tag} remove={removeTag}/>
            })}
          </td>
        </tr>
    );
  }
}
