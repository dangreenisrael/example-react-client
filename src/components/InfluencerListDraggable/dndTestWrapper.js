import TestBackend from 'react-dnd-test-backend';
import { DragDropContext } from 'react-dnd';
import React, {Component} from 'react';
import { Provider } from 'react-redux';
import store from '../../redux/store';

export default (InnerComponent) => {
  @DragDropContext(TestBackend)
  class TestContextContainer extends Component {
    render() {
      return <Provider store={store}>{InnerComponent}</Provider>;
    }
  }
  return TestContextContainer;
}