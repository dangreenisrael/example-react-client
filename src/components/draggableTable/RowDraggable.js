import React, {Component} from 'react';
import {DragSource} from 'react-dnd';
import CheckboxRowContainer from './CheckboxRowContainer';
const boxSource = {
  beginDrag(props) {
    return { id: props.id };
  },
};

@DragSource(props => "application", boxSource, (connect, monitor) => ({
  connectDragSource: connect.dragSource(),
  connectDragPreview: connect.dragPreview(),
  isDragging: monitor.isDragging(),
}))

export default class DraggableRow extends Component {
  constructor(props){
    super(props);
    this.toggleChecked = this.toggleChecked.bind(this);
    this.state = {checked: false};
  }
  toggleChecked(status){
    this.setState({checked: status});
  }
  render() {
    this.props.connectDragPreview(this.props.dragPreview);
    const {connectDragSource, data, headings, id} = this.props;
    const style = this.state.checked ? {backgroundColor: "rgba(23, 42, 251, 0.34)" } : {};
    const cellData = headings.map(key =>{
      return data[key]
    });
    return connectDragSource(
        <tr style={style}>
          <td>
            <CheckboxRowContainer rowId={id} toggleChecked={this.toggleChecked}/>
          </td>
          {cellData.map((cell, i)=>{
            return <td key={i}>
              {cell}
            </td>
          })}
        </tr>
    );
  }
}
