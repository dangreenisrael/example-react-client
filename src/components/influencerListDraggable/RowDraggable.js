import React, {Component} from 'react';
import {DragSource} from 'react-dnd';
import RowCheckboxContainer from './RowCheckboxContainer';
import Tag from '../Tag'

const boxSource = {
  beginDrag(props) {
    return { id: props.id };
  },
};

@DragSource(props => "application", boxSource, (connect, monitor) => ({
  connectDragSource: connect.dragSource(),
  connectDragPreview: connect.dragPreview(),
  isDragging: monitor.isDragging(),
}))
export default class DraggableRow extends Component {
  render() {
    this.props.connectDragPreview(this.props.dragPreview);
    const {connectDragSource, data, headings, id, selected, removeTag} = this.props;
    const style = selected ? {backgroundColor: "rgba(23, 42, 251, 0.34)" } : {};
    const cellData = headings.map(key =>{
      return data[key]
    });
    return connectDragSource(
        <tr style={style}>
          <td>
            <RowCheckboxContainer rowId={id}/>
          </td>
          {cellData.map((cell, i)=>{
            return <td key={`data-${i}`}>
              {cell}
            </td>
          })}
          <td>
            {data.tags.map((tag, i)=>{
              return <Tag key={`tags-${i}`} title={tag} remove={removeTag}/>
            })}
          </td>
        </tr>
    );
  }
}
