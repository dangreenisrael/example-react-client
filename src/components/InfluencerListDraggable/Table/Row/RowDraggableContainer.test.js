import React from 'react';
import renderer from 'react-test-renderer';
import RowDraggable from './RowDraggable';
import dndTestWrapper from '../../dndTestWrapper';

const headings = ["firstName", "lastName", "userName"];
const influencer = {
  firstName: 'Dan',
  lastName: 'Green',
  userName: 'DGreen',
  tags: []
};

test('Row Draggable Renders correctly', () => {
  const RowDraggableWrapper = dndTestWrapper(
      <RowDraggable name='test'
                    headings={headings}
                    data={influencer}
      />);
  const tree = renderer.create(<RowDraggableWrapper/>).toJSON();
  expect(tree).toMatchSnapshot();
});
