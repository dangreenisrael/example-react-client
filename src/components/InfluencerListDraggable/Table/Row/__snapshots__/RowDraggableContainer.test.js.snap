exports[`test Row Draggable Renders correctly 1`] = `
<tr
  style={Object {}}>
  <td>
    <input
      checked={false}
      name={undefined}
      onChange={[Function]}
      type="checkbox" />
  </td>
  <td
    style={undefined} />
  <td
    style={undefined} />
  <td
    style={undefined} />
  <td />
</tr>
`;
